tests/rptest/tests/acls_test.py

```diff
@@ -315,6 +315,103 @@ def test_mtls_principal(self, rules=None, fail=False):
             except ClusterAuthorizationError:
                 assert fail, "list acls should have succeeded"
 
+    @cluster(num_nodes=3)
+    @parametrize(authn_method='sasl')
+    @parametrize(authn_method='mtls_identity')
+    def test_security_feature_migration(self, authn_method: str):
+        def check_perms():
+            try:
+                self.get_client("base").acl_list()
+                self.logger.debug('list acls should have failed for base user')
+                return False
+            except ClusterAuthorizationError:
+                pass
+
+            # should succeed
+            try:
+                self.get_client("cluster_describe").acl_list()
+            except Exception:
+                self.logger.debug(
+                    'list acls should have passed for cluster user')
+                return False
+
+            try:
+                self.get_super_client().acl_list()
+            except Exception:
+                self.logger.debug(
+                    'list acls should have passed for super user')
+                return False
+
+            return True
+
+        self.prepare_cluster(use_tls=True, use_sasl=True)
+
+        wait_until(check_perms,
+                   timeout_sec=90,
+                   err_msg='check_perms failed before migration')
+
+        # Change the authn method and initiate feature migration
+        # with a rolling restart. Do not recreate tls certs.
```

Comment:
it would be nice if this comment described why the certs were not recreated. also, if the authn method is being changed, why would it matter if the certs were recreated?

Reply:
In services/tls.py there is the following assert:
```python
    def create_cert(self, .. args ..):
        name = name or host
        assert name not in self.certs, f"cert '{name}' already exists"  # <--- assert here
```
which fails when we re-create the certs. I assumed someone put that assert in for a good reason so I added the flag to control cert construction.

Reply:
I'll add that into the comment.

Reply:
I'll touch base with the original tls.py author to see why that assert is necessary.

Reply:
This is a defensive assertion. The assertion exists not to prevent certificate recreation, but because the implementation didn't take that scenario into account when it was originally built.

Reply:
> This is a defensive assertion. The assertion exists not to prevent certificate recreation, but because the implementation didn't take that scenario into account when it was originally built.

Thanks. I changed the commit to remove that assert and keep cert construction.